src/libs/OptionsListUtils.ts

```diff
@@ -1370,7 +1371,9 @@ function getTaxRatesSection(taxRates: TaxRatesWithDefault | undefined, selectedO
     const taxes = transformedTaxRates(taxRates);
 
     const sortedTaxRates = sortTaxRates(taxes);
-    const enabledTaxRates = sortedTaxRates.filter((taxRate) => !taxRate.isDisabled);
+    const enabledTaxRates = sortedTaxRates
+        .filter((taxRate) => !taxRate.isDisabled)
+        .map((taxRate) => ({...taxRate, isSelected: selectedOptions.some((option) => option.name === taxRate.modifiedName)}));
```

Comment:
Can we compare id's here instead?

Reply:
@MonilBhavsar we don't have id in `selectedOptions` as we constructed it only from `selectedTax` here
https://github.com/Expensify/App/blob/34434a54704ff8e4df4929b9e51a39f29dfe131f/src/components/TaxPicker.tsx#L49-L59

Reply:
Can we add taxCode there and compare it?

Reply:
@MonilBhavsar there are different places we use tax picker and not straightforward to get `taxCode` for instance here
https://github.com/Expensify/App/blob/8fbf52e65986e02298bb5fb5a2a4a06f65637c21/src/pages/iou/request/step/IOURequestStepTaxRatePage.tsx#L63-L64
there is not taxCode in `transaction.taxRate` (for creating request in confirmation page)
`
{
  "text": "Tax 3 (3%) • Default",
  "keyForList": "Tax 3 (3%) • Default",
  "searchText": "Tax 3 (3%) • Default",
  "tooltipText": "Tax 3 (3%) • Default",
  "isDisabled": false,
  "data": {
    "modifiedName": "Tax 3 (3%) • Default",
    "isDisabled": false
  },
  "isSelected": true
}
`
Maybe we need some other refactoring pr and I don't think it would be appropriate to handle in the scope of this pr, in terms of what we are trying to fix here

Reply:
bump